lib/inc/internal/detectors/vmware_detector.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <internal/detectors/result.hpp>
+#include <internal/sources/cpuid_source.hpp>
+#include <internal/sources/dmi_source.hpp>
+#include <string>
+
+namespace whereami { namespace detectors {
+
+    /**
+     * VMware detector function
+     * @param cpuid_source An instance of a CPUID data source
+     * @param dmi_source An instance of a DMI data source
+     * @return The VMware detection result
+     */
+    result vmware(const sources::cpuid_base& cpuid_source,
+                  const sources::dmi_base& dmi_source);
+
+    /**
+     * Retrieve the VMware version based on a DMI BIOS address result, if possible
+     * @param address The value of the BIOS address from DMI
+     * @return The VMware version
+     */
+    std::string vmware_bios_address_to_version(int address);
```

Comment:
Since this doesn't need to be called outside of the vmware detector, it can just be a static method in that file - there's no need to put it in the header or even in the namespace.